setup.py

```diff
@@ -1,36 +1,35 @@
-import io
 import os
-from setuptools import setup, find_packages
 
 import versioneer
+from setuptools import find_packages, setup
 
 PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
 
-DESCRIPTION = 'Python module and CLI for hashing of file system directories.'
+DESCRIPTION = "Python module and CLI for hashing of file system directories."
 
 try:
-    with io.open(os.path.join(PROJECT_ROOT, 'README.md'), encoding='utf-8') as f:
-        long_description = '\n' + f.read()
-except IOError:
+    with open(os.path.join(PROJECT_ROOT, "README.md"), encoding="utf-8") as f:
+        long_description = "\n" + f.read()
+except OSError:
     long_description = DESCRIPTION
 
 setup(
-    name='dirhash',
+    name="dirhash",
     version=versioneer.get_version(),
     cmdclass=versioneer.get_cmdclass(),
     description=DESCRIPTION,
     long_description=long_description,
     long_description_content_type="text/markdown",
-    url='https://github.com/andhus/dirhash-python',
+    url="https://github.com/andhus/dirhash-python",
     author="Anders Huss",
     author_email="andhus@kth.se",
-    license='MIT',
-    install_requires=['scantree>=0.0.2'],
-    packages=find_packages('src'),
-    package_dir={'': 'src'},
+    license="MIT",
+    install_requires=["scantree"],
+    packages=find_packages("src"),
+    package_dir={"": "src"},
     include_package_data=True,
     entry_points={
-        'console_scripts': ['dirhash=dirhash.cli:main'],
+        "console_scripts": ["dirhash=dirhash.cli:main"],
     },
-    tests_require=['pytest', 'pytest-cov']
+    tests_require=["pre-commit", "pytest", "pytest-cov"],
```

Comment:
manual change - adding pre-commit

Reply:
Generally, I think it is better to install `pre-commit` separately from your main project venv. Doesn't matter much here, but is it really needed in `tests_require`, or just for convenience? (In tox/workflows it is installed explicitly anyway, right?)

Reply:
yes, but having discrepancies between requirements in workflows/tox and setup.py usually lead to issues down the line ... i usually adjust my projects so that they have one source of truth for the dependencies and processes are looking there ... might try it in separate PR